src/org/omegat/Main.java

```diff
@@ -417,33 +420,39 @@ private static int runConsoleStats() throws Exception {
         }
 
         String outputFilename = PARAMS.get(CLIParameters.STATS_OUTPUT);
-        String statsMode;
+        StatOutputFormat statsMode;
         if (PARAMS.containsKey(CLIParameters.STATS_MODE)) {
-            statsMode = PARAMS.get(CLIParameters.STATS_MODE);
+            statsMode = StatOutputFormat.parse(PARAMS.get(CLIParameters.STATS_MODE));
         } else {
-            // when no stats type specified, try to detect from file extension, otherwise XML.
-            if (outputFilename.endsWith(".json") || outputFilename.endsWith(".JSON")) {
-                statsMode = "JSON";
-            } else if (outputFilename.endsWith(".xml") || outputFilename.endsWith(".XML")) {
-                statsMode = "XML";
-            } else if (outputFilename.endsWith(".txt") || outputFilename.endsWith(".TXT")) {
-                statsMode = "TXT";
+            // when no stats type specified, try to detect from file extension,
+            // otherwise XML.
+            if (outputFilename.toLowerCase().endsWith(StatOutputFormat.JSON.getFileExtension())) {
+                statsMode = StatOutputFormat.JSON;
+            } else if (outputFilename.toLowerCase().endsWith(StatOutputFormat.XML.getFileExtension())) {
+                statsMode = StatOutputFormat.XML;
+            } else if (outputFilename.toLowerCase().endsWith(StatOutputFormat.TEXT.getFileExtension())) {
+                statsMode = StatOutputFormat.TEXT;
             } else {
-                statsMode = "XML";
+                statsMode = StatOutputFormat.XML;
```

Comment:
```suggestion
                statsMode = StatOutputFormat.JSON;
```

If we want to have all JSON stats by default, but it could break scripts/commands that use the CLI to produce stats that are expecting an XML output (current default for CLI).